models/classes/Lists/DataAccess/Repository/DependentPropertiesRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\tao\model\Lists\DataAccess\Repository;
+
+use core_kernel_classes_Property;
+use oat\generis\model\GenerisRdf;
+use oat\generis\model\OntologyRdf;
+use oat\oatbox\service\ConfigurableService;
+use oat\search\helper\SupportedOperatorHelper;
+use oat\tao\model\Lists\Business\Domain\DependentPropertiesRepositoryContext;
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+use oat\tao\model\Lists\Business\Contract\DependentPropertiesRepositoryInterface;
+
+class DependentPropertiesRepository extends ConfigurableService implements DependentPropertiesRepositoryInterface
+{
+    /**
+     * @return core_kernel_classes_Property[]
+     */
+    public function findAll(DependentPropertiesRepositoryContext $context): array
```

Comment:
Some points: 

- You should use @inheritDoc instead of maintaining the annotations in multiple places.
- Use ContextInterface for better open/closed compatibility.

```suggestion
    /**
     * @inheritDoc
     */
    public function findAll(ContextInterface $context): array
```

Reply:
Already did this on my local 😅 
But thanks for reminding me